src/software/new_geom/circle.cpp

```diff
@@ -0,0 +1,55 @@
+#include "software/new_geom/circle.h"
+
+Circle::Circle() : origin_(Point()), radius_(0) {}
+
+Circle::Circle(const Point &origin, double radius) : origin_(origin)
+{
+    setRadius(radius);
+}
+
+void Circle::setOrigin(const Point &o)
+{
+    origin_ = o;
+}
+
+Point Circle::getOrigin() const
+{
+    return origin_;
+}
+
+void Circle::setRadius(double radius)
+{
+    if (radius < 0)
+    {
+        throw std::invalid_argument("Circle radius cannot be negative, given: " +
+                                    std::to_string(radius));
+    }
+
+    radius_ = radius;
+}
+
+double Circle::getRadius() const
+{
+    return radius_;
+}
+
+bool Circle::contains(const Point &p) const
+{
+    return p.distanceFromPoint(origin_) <= radius_;
```

Comment:
Why would we need `EPSILON` with a `<=` operator?

Reply:
We use ```EPSILON``` in ```operator==``` to check if the radii of two circles is close enough. For this case, we are just checking that the distance of ```p``` from the center of the circle is within the circle ```radius```

Reply:
It seems inconsistent to say that when comparing radii we can be off by EPSILON, but when comparing a radius to a distance, we can't be off by EPSILON.

Reply:
`EPSILON` should be used to test equality, but here we are testing if the distance should be less than or equal to the radius. Not sure why `EPSILON` is necessary in this case.

Reply:
We can have a case of two circles being `==` to each other and yet `contain` different points, namely points one EPSILON outside of the smaller circle That's the inconsistency that I'm concerned about

Reply:
I think that inconsistency can still happen if you do a comparison like `p.distanceFromPoint(origin_)  - radius_ <= EPSILON;`, since what the EPSILON does in this case is adding a bit of a buffer for checking of a `Point` is in the `Circle`; If two `Circle`s have a slight mismatch that still qualifies them as being equal to each other, a `Point` still be in `radius_ + EPSILON` for one `Circle` but not the other

Reply:
@jonathanlew - what are you imagining this looking like with `EPSILON`?

Reply:
@jonathanlew did you still have concerns about this?

Reply:
Nothing blocking here. There's potentially a better way to handle this (out of scope for this PR), but the current implementation seems consistent with what we've done in the past